src/ngx_http_lua_pipe.c

```diff
@@ -0,0 +1,2476 @@
+static ngx_int_t ngx_http_lua_pipe_init_ctx(
+    ngx_http_lua_pipe_ctx_t **pipe_ctx_pt, int fd, ngx_pool_t *pool,
+    u_char *errbuf, size_t *errbuf_size);
+static ngx_int_t ngx_http_lua_pipe_write(ngx_http_lua_pipe_t *pipe,
+    ngx_http_lua_pipe_ctx_t *pipe_ctx);
+static int ngx_http_lua_pipe_read_stdout_retval(
+    ngx_http_lua_ffi_pipe_proc_t *proc, lua_State *L);
+static int ngx_http_lua_pipe_read_stderr_retval(
+    ngx_http_lua_ffi_pipe_proc_t *proc, lua_State *L);
+static int ngx_http_lua_pipe_read_retval_helper(
+    ngx_http_lua_ffi_pipe_proc_t *proc, lua_State *L, int from_stderr);
+static int ngx_http_lua_pipe_write_retval(ngx_http_lua_ffi_pipe_proc_t *proc,
+    lua_State *L);
+static int ngx_http_lua_pipe_wait_retval(ngx_http_lua_ffi_pipe_proc_t *proc,
+    lua_State *L);
+static void ngx_http_lua_pipe_resume_helper(ngx_event_t *ev,
+    ngx_http_lua_co_ctx_t *wait_co_ctx);
+static void ngx_http_lua_pipe_resume_read_stdout_handler(ngx_event_t *ev);
+static void ngx_http_lua_pipe_resume_read_stderr_handler(ngx_event_t *ev);
+static void ngx_http_lua_pipe_resume_write_handler(ngx_event_t *ev);
+static void ngx_http_lua_pipe_resume_wait_handler(ngx_event_t *ev);
+static ngx_int_t ngx_http_lua_pipe_resume(ngx_http_request_t *r);
+static void ngx_http_lua_pipe_dummy_event_handler(ngx_event_t *ev);
+static void ngx_http_lua_pipe_cleanup_helper(
+    ngx_http_lua_co_ctx_t *wait_co_ctx);
+static void ngx_http_lua_pipe_proc_read_stdout_cleanup(void *data);
+static void ngx_http_lua_pipe_proc_read_stderr_cleanup(void *data);
+static void ngx_http_lua_pipe_proc_write_cleanup(void *data);
+static void ngx_http_lua_pipe_proc_wait_cleanup(void *data);
+
+
+static ngx_rbtree_t       ngx_http_lua_pipe_rbtree;
+static ngx_rbtree_node_t  ngx_http_lua_pipe_proc_sentinel;
+
+
+#if (NGX_HTTP_LUA_HAVE_SIGNALFD)
+static int                                ngx_http_lua_signalfd;
+static struct signalfd_siginfo            ngx_http_lua_pipe_notification;
+
+#define ngx_http_lua_read_sigfd           ngx_http_lua_signalfd
+
+#else
+static int                                ngx_http_lua_sigchldfd[2];
+static u_char                             ngx_http_lua_pipe_notification[1];
+
+#define ngx_http_lua_read_sigfd           ngx_http_lua_sigchldfd[0]
+#define ngx_http_lua_write_sigfd          ngx_http_lua_sigchldfd[1]
+#endif
+
+
+static ngx_connection_t                  *ngx_http_lua_sigfd_conn = NULL;
+
+
+#define SIG(name)  { (SIG##name),  "SIG"#name }
+
+
+/* signals below are handled specially in Nginx.
+ * We need to reset them for the spawned child processes */
+ngx_http_lua_pipe_signal_t ngx_signals[] = {
+    SIG(SYS),
+    SIG(PIPE),
+    { 0, NULL }
+};
+
+
+enum {
+    PIPE_ERR_CLOSED = 1,
+    PIPE_ERR_SYSCALL,
+    PIPE_ERR_NOMEM,
+    PIPE_ERR_TIMEOUT,
+    PIPE_ERR_ADD_READ_EV,
+    PIPE_ERR_ADD_WRITE_EV
+};
+
+
+enum {
+    PIPE_READ_ALL     = 0,
+    PIPE_READ_BYTES   = 1,
+    PIPE_READ_LINE    = 2,
+    PIPE_READ_ANY     = 3
```

Comment:
Ditto, this enum's values could be anonymous as well. Besides, aligning the assignment operator typically makes for messy diffs in the future. But the problem doesn't even arise itself if we let those values (or even all but the first one) be anonymously defined. So that's one more reason.